Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalTBModuleX.cc

```diff
@@ -0,0 +1,378 @@
+  }
+#endif
+  const auto& layerType = args.value<std::vector<int> >("LayerType");    // Type of the layer
+  const auto& layerSense = args.value<std::vector<int> >("LayerSense");  // Content of a layer
+  const auto& maxModule = args.value<std::vector<int> >("MaxModule");    // Maximum # of row/column
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HGCalGeom") << "DDHGCalTBModuleX: " << layerType.size() << " layers";
+  for (unsigned int i = 0; i < layerType.size(); ++i)
+    edm::LogVerbatim("HGCalGeom") << "Layer [" << i << "] with material type " << layerType[i] << " sensitive class "
+                                  << layerSense[i] << " and " << maxModule[i] << " maximum row/columns";
+#endif
+  const auto& zMinBlock = args.value<double>("zMinBlock");  // Starting z-value of the block
+  const auto& rMaxFine = args.value<double>("rMaxFine");    // Maximum r-value for fine wafer
+  const auto& waferW = args.value<double>("waferW");        // Width of the wafer
+  const auto& waferGap = args.value<double>("waferGap");    // Gap between 2 wafers
+  const auto& absorbW = args.value<double>("absorberW");    // Width of the absorber
+  const auto& absorbH = args.value<double>("absorberH");    // Height of the absorber
+  const auto& rMax = args.value<double>("rMax");            // Maximum radial extent
+  const auto& rMaxB = args.value<double>("rMaxB");          // Maximum radial extent of a block
+  double waferTot = waferW + waferGap;
+  std::string idName = DDSplit(args.parentName()).first;
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HGCalGeom") << "DDHGCalTBModuleX: zStart " << convertCmToMm(zMinBlock) << " rFineCoarse "
+                                << convertCmToMm(rMaxFine) << " wafer width " << convertCmToMm(waferW)
+                                << " gap among wafers " << convertCmToMm(waferGap) << " absorber width "
+                                << convertCmToMm(absorbW) << " absorber height " << convertCmToMm(absorbH) << " rMax "
+                                << convertCmToMm(rMax) << ":" << convertCmToMm(rMaxB);
+  edm::LogVerbatim("HGCalGeom") << "DDHGCalTBModuleX: NameSpace " << ns.name() << " Parent Name " << idName;
+  std::unordered_set<int> copies;  // List of copy #'s
+  copies.clear();
+#endif
+
+  dd4hep::Volume parent = ns.volume(args.parentName());
+  double zi(zMinBlock);
+  for (unsigned int i = 0; i < blockThick.size(); i++) {
+    double zo = zi + blockThick[i];
+    std::string name = idName + "Block" + std::to_string(i);
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalTBModuleX: Block " << i << ":" << name << " z " << convertCmToMm(zi) << ":"
+                                  << convertCmToMm(zo) << " R " << convertCmToMm(rMaxB) << " T "
+                                  << convertCmToMm(blockThick[i]);
+#endif
+    dd4hep::Material matter = ns.material(genMat);
+    dd4hep::Solid solid = dd4hep::Tube(0.5 * blockThick[i], 0, rMaxB, 0.0, 2._pi);
```

Comment:
```
//  Tube(rmin, rmax, dz, startPhi, startPhi + deltaPhi)
-    dd4hep::Solid solid = dd4hep::Tube(0.5 * blockThick[i], 0, rMaxB, 0.0, 2._pi);
+    dd4hep::Solid solid = dd4hep::Tube(0, rMaxB, 0.5 * blockThick[i], 0.0, 2._pi);
```
@bsunanda - The Tube API has changed and this one was causing the voxelization problem:
```
----- Begin Fatal Exception 16-Sep-2019 10:25:05 CEST-----------------------
An exception of category 'StdException' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'p'
   [2] Calling method for module DDTestDumpFile/'testDump'
   [3] Using EventSetup component DDDetectorESProducer/'' to make data DDDetector/'DDHGCalTBModuleX' in record GeometryFileRcd
Exception Message:
A std::exception was thrown.
An exception of category 'FatalRootError' occurred.
   Additional Info:
      [a] Fatal Root Error: @SUB=TGeoVoxelFinder::SortAll
Cannot voxelize HGCalEE :less than 2 boundaries on X


dd4hep: while parsing /build/yana/DD4HEP/Sunanda/CMSSW_11_0_X_2019-09-12-2300/src/Geometry/HGCalCommonData/data/dd4hep/cms-test-ddhgcalTBModuleX-algorithm.xml
dd4hep: with plugin:DD4hep_CompactLoader
----- End Fatal Exception -------------------------------------------------
```

Reply:
Thanks Yana - made the correction